### DISCUSSION TITLE
Pull Push not working on server

### DISCUSSION THREAD

**Discussion by robin-vmipl** [community/community#130543] [Other Feature Feedback, Questions, & Ideas]
I am writing to seek assistance with an issue we are experiencing on one of our servers where we host our code using Git. For the past few days, we have been unable to perform git pull and git push operations from this server. These operations were previously working without any issues.

Every time we attempt to execute a git pull or git push, we encounter a timeout error. Additionally, we have noticed that we cannot ping github.com from this server, although we can successfully ping other sites such as google.com.

Here are some details regarding our troubleshooting steps:

DNS Resolution: nslookup and dig commands for github.com fail to resolve.
Network Configuration: We have verified that other servers and local machines on the same network do not encounter this issue.
Firewall and Security: We have checked our firewall rules and found no configurations that should be blocking access to GitHub.
Traceroute: traceroute github.com shows the connection failing at a certain point.
Given that this issue is specific to github.com and not other sites, we suspect there might be a block or some specific network issue affecting our server's connection to GitHub.

**Comment by luckyshark1012** [comment]
# Check DNS Resolution
nslookup github.com
dig github.com

# Check Network Configuration
ip addr
ip route

# Check Firewall Rules
sudo ufw status
sudo iptables -L

# Traceroute
traceroute github.com
By systematically working through these steps, you should be able to identify and resolve the issue preventing your server from accessing GitHub. If all else fails, contacting GitHub support with detailed diagnostic information would be a prudent next step.
